src/xmipp/applications/tests/function_tests/test_metadata_db_main.cpp

```diff
@@ -61,17 +67,46 @@ class MetadataTest : public ::testing::Test
 
     // virtual void TearDown() {}//Destructor
 
-    MetaData mDsource,mDanotherSource;
-    MetaData mDunion, mDjoin;
+    MetaDataDb mDsource,mDanotherSource;
+    MetaDataDb mDunion, mDjoin;
     size_t id, id1,id2;
+    std::vector<int> mDsourceIds;
 };
 
+TEST_F( MetadataTest, IdIteration)
+{
+    auto it = mDsource.ids().begin();
+    for (size_t i = 0; i < mDsourceIds.size(); i++, ++it);
+    ASSERT_EQ(it, mDsource.ids().end());
+
+    size_t i = 0;
+    for (size_t objId : mDsource.ids()) {
+        ASSERT_EQ(objId, mDsourceIds[i]);
+        i++;
+    }
+    ASSERT_EQ(i, mDsourceIds.size());
+}
+
+TEST_F( MetadataTest, RowIteration)
+{
+    auto it = mDsource.begin();
```

Comment:
not implemented

Reply:
Sorry, I forgot to decomment code. Done.